### PR TITLE
relax zonename validator; enhance handling of base attributes

### PR DESCRIPTION
**Pull request by hadfl**
fixes https://github.com/omniosorg/zadm/issues/110

### CODE REVIEW

File: lib/Zadm/Schema/base.pm [hunk 1]
```diff
@@ -12,7 +12,7 @@ has schema => sub($self) {
     return {
     zonename    => {
         description => 'name of zone',
-        validator   => $self->sv->regexp(qr/^[-\w]+$/, 'zonename not valid'),
+        validator   => $self->sv->regexp(qr/^.+$/, 'expected a string'),
```
Comment (oetiker) [inline lib/Zadm/Schema/base.pm:15]: hmmm are you allowing zone names consisting of space on purpose? how about `^\S(.*\S)?$`

File: lib/Zadm/Schema/base.pm [hunk 2]
```diff
@@ -27,22 +27,22 @@ has schema => sub($self) {
...
         description => 'the maximum set of privileges any process in this zone can obtain',
         default     => 'default',
-        validator   => $self->sv->regexp(qr/^[-+!\w,]+$/, 'limitpriv not valid'),
+        validator   => $self->sv->regexp(qr/^[-+!\w,]*$/, 'limitpriv not valid'),
     },
     brand       => {
         description => "the zone's brand type",
         default     => 'lipkg',
-        validator   => sub { return undef },
+        validator   => $self->sv->regexp(qr/^.+$/, 'expected a string'),
```
Comment (oetiker) [inline lib/Zadm/Schema/base.pm:45]: brand names consisting of spaces ?

Comment (hadfl) [reply lib/Zadm/Schema/base.pm:45]: the brand names are checked when `zadm` gets called. this has just been changed for consistency with other validators

File: lib/Zadm/Schema/base.pm (hunk 1 above)
Comment (hadfl) [reply lib/Zadm/Schema/base.pm:15]: good point, thanks

**Review by citrus-it — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**